PocketKit/Sources/PocketKit/Authorization/AuthorizationClient.swift

```diff
@@ -22,11 +24,25 @@ public class AuthorizationClient {
 
     @MainActor
     func logIn(from contextProvider: ASWebAuthenticationPresentationContextProviding?) async throws -> Response {
+        defer { isAuthenticating = false }
+
+        guard isAuthenticating == false else {
+            throw AuthorizationClient.Error.alreadyAuthenticating
+        }
+
+        isAuthenticating = true
```

Comment:
looks good, just curious - any reason why we defer first here and defer after the guard in `signUp`

Reply:
Just preference - to me, it just makes it more obvious that it's happening. Though, in theory, this `defer` does nothing if the guard fails since the value will already be `false`.